packages/react/src/components/Geo/LocationSearch/index.tsx

```diff
@@ -17,6 +17,7 @@ const LOCATION_SEARCH_CONTAINER = 'geocoder-container';
 
 type AmplifyLocationSearch = IControl & {
   addTo: (container: string) => void;
+  on: (event: string, fn: Function) => void;
```

Comment:
For readability/ease of use:

- Can we rename the `event` arg to `eventType`, and narrow the values passed allowed to a string union of potential event types (example `'clear'`, `'result'`) in place of `string`. 
- Rename `fn` to `callback` and update `Function` with the type signature of the expected by `on`, for example `(event: UnderlyingEventType) => void`.

Reply:
Just having some issues figuring out the underlying event types for each function.

Reply:
Added some type information, any advice on the rest would be welcome